.github/workflows/create-release-main.yml

```diff
@@ -0,0 +1,71 @@
+name: Create new preview release (main)
+
+on:
+  # run when pushed to main is published,
+  # which creates a new tag
+  push:
+    branches:
+      - main
+
+env:
+  ARTIFACT_VERSION: experimental
+
+jobs:
+  build-and-push:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0 # required to access tags
+          submodules: "true"
+
+      - name: Build & run devcontainer image
+        # this always builds a new image from scratch rather than from the build-devcontainer-image workflow output
+        # so that we pick up the latest versions of everything
+        # NB: if you update this also update live-validation.yml and create-release-official.yaml
+        id: devcontainer
+        run: |
+          docker build --tag devcontainer:latest .devcontainer
+          mkdir -p $HOME/.docker # in case it doesn't exist
+          container_id=$(docker create -w /workspace -v $GITHUB_WORKSPACE:/workspace -v /var/run/docker.sock:/var/run/docker.sock devcontainer:latest)
+          docker start "$container_id"
+          echo "container_id=$container_id" >> $GITHUB_ENV
+
+      - name: Build required release files
+        run: |
+          container_id=${{env.container_id}}
+          docker exec "$container_id" task VERSION=${{ env.ARTIFACT_VERSION }} LATEST_VERSION_TAG=${{ env.ARTIFACT_VERSION }} make-release-artifacts
+
+      - name: Upload release assets
+        uses: svenstaro/upload-release-action@7319e4733ec7a184d739a6f412c40ffc339b69c7 # this is v2.5.0, but pinned
+        with:
+          repo_token: ${{ secrets.GITHUB_TOKEN }}
+          tag: refs/tags/${{ env.ARTIFACT_VERSION }}
+          file: "v2/out/release/*"
+          file_glob: true
+          prerelease: true
+          overwrite: true
+          release_name: "Experimental"
+          body: |
+            This is the "latest" releasewhich contains the most recent commits from the main branch. (commit: ${{ github.sha }}).
```

Comment:
I think we should avoid using the term _latest_ as that may be misinterpreted.

```suggestion
            This is an experimental release which containing the most recent commits from the main branch as of commit: ${{ github.sha }}.
```

Also, is there a convenient timestamp we could include in this message as well, so it's easy for a reader to determine whether it's stale or not?

Reply:
That is why I added the commit, but will see if I can add a timestamp although it would be the release's date:
![image](https://github.com/Azure/azure-service-operator/assets/4345663/34fb887d-f65e-4744-88bb-fc16675a1f84)

Verifying with a new run in my fork to see if those are properly updated. (one would expect)

Reply:
Release does not get updated, release assets do:
![Uploading image.png…]()

Reply:
Good enough. Thanks for checking.